scrapy_poet/commands.py

```diff
@@ -93,9 +105,9 @@ def run(self, args, opts):
         frozen_time = datetime.datetime.now(datetime.timezone.utc).replace(
```

Comment:
Are we rewinding the time back, because frozen_time is frozen before the spider runs? Would it be better to compute frozen_time in the callback as well?

Reply:
I think it doesn't matter where is the value calculated (it shouldn't matter what is the actual value as long as it's saved into meta), but I think it's easier to pass the value into the spider than outside it?

Reply:
It's very minor, but there could be some strange cases. 

For example, you downloaded the response, and response contains time information. In the current implementation we freeze time only after the download, but the frozen_time is set before the download, so they don't match; it means that the body will contain time which is in future. If you want to do something with this time, you may get negative offsets, or it may cause some other issues. I don't know, maybe there is some http header (like Expires), which would contain incorrect information with the current implementation. 

Again, quite an edge case, and it's just seconds, but the right thing to do seems to be to have frozen_time to match the time on the clock.

Reply:
Yeah, makes sense. Though we still need to remove the microseconds part in case the time ticks by some small sub-second amount but the seconds value is still increased.